crates/matrix-sdk-search/src/index.rs

```diff
@@ -372,4 +387,85 @@ mod tests {
 
         Ok(())
     }
+
+    #[test]
+    fn test_redaction_removes_event() -> Result<(), Box<dyn Error>> {
+        let room_id = room_id!("!room_id:localhost");
+        let mut index =
+            RoomIndex::new_in_memory(room_id).expect("failed to make index in ram: {index:?}");
+
+        let event_id = event_id!("$event_id:localhost");
+        let user_id = user_id!("@user_id:localhost");
+        let event = EventFactory::new()
+            .text_msg("This is a sentence")
+            .event_id(event_id)
+            .room(room_id)
+            .sender(user_id)
+            .into_any_sync_message_like_event();
+
+        index.handle_event(event)?;
+
+        index.commit_and_reload()?;
+
+        assert!(index.contains(event_id), "Index should contain event");
+
+        let redaction_event_id = event_id!("$redaction_event_id:localhost");
+        let redaction = EventFactory::new()
+            .redaction(event_id)
+            .event_id(redaction_event_id)
+            .room(room_id)
+            .sender(user_id)
+            .into_any_sync_message_like_event();
+
+        index.handle_event(redaction)?;
+
+        index.commit_and_reload()?;
+
+        assert!(!index.contains(event_id), "Index should not contain event");
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_edit_removes_old_and_adds_new_event() -> Result<(), Box<dyn Error>> {
+        let room_id = room_id!("!room_id:localhost");
+        let mut index =
+            RoomIndex::new_in_memory(room_id).expect("failed to make index in ram: {index:?}");
+
+        let old_event_id = event_id!("$old_event_id:localhost");
+        let user_id = user_id!("@user_id:localhost");
+        let old_event = EventFactory::new()
+            .text_msg("This is a sentence")
+            .event_id(old_event_id)
+            .room(room_id)
+            .sender(user_id)
+            .into_any_sync_message_like_event();
+
+        index.handle_event(old_event)?;
+
+        index.commit_and_reload()?;
+
+        assert!(index.contains(old_event_id), "Index should contain event");
+
+        let new_event_id = event_id!("$new_event_id:localhost");
+        let edit = EventFactory::new()
```

Comment:
ditto, let's reuse the `EventFactory`, and use fixed `room`/`sender`s on it, to make the test less verbose.